### DISCUSSION TITLE
SkyWalking Data Storage Questions

### DISCUSSION THREAD

**Discussion by AllenArbiter** [apache/skywalking#11412] [General]
Hello everyone,

I have been using SkyWalking for performance monitoring, but I have some questions about its data storage strategy and would appreciate some answers. Here are my questions:

Why does SkyWalking store Counter metric data every minute? Is this frequency configurable, or are there specific reasons for storing data every minute?

I am concerned about the accumulation of a large volume of data when running SkyWalking for an extended period, which could potentially deplete storage resources. Therefore, I would like to understand SkyWalking's data storage strategy. Are there any data management policies in place to address long-term storage challenges?

These questions are essential for gaining a better understanding of how SkyWalking manages data and its storage approach, as well as for planning storage resources. If you have experience with SkyWalking or insights into its internal workings, please feel free to share your thoughts. Thank you for your assistance!

I hope this helps you draft your questions effectively on GitHub. You can further refine the wording as needed to accurately express your concerns and requirements.

**Comment by wu-sheng** [comment]
> Why does SkyWalking store Counter metric data every minute? Is this frequency configurable, or are there specific reasons for storing data every minute?

Do you prefer higher accuracy or lower? We picked that as usually it is enough for system operation, and costing memory/disk reasonable. 

> Therefore, I would like to understand SkyWalking's data storage strategy. Are there any data management policies in place to address long-term storage challenges?

This is called TTL in SkyWalking. By default, you have 7 days metrics and 3 days traces/logs. Ref doc, https://skywalking.apache.org/docs/main/v9.6.0/en/setup/backend/ttl/#ttl
